File: src/main/java/fr/univbrest/dosi/spi/service/QuestionService.java
package fr.univbrest.dosi.spi.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import fr.univbrest.dosi.spi.bean.Etudiant;
import fr.univbrest.dosi.spi.bean.Question;
import fr.univbrest.dosi.spi.dao.*;


@Service
public class QuestionService {
	
	@Autowired
	private  QuestionRepository QstRepository;
	
    public final List<Question> getAll(){
		
		return (List<Question>) QstRepository.findAll();
	}
	
	public final Question updateQuestion(final Question Qst){
		return QstRepository.save(Qst);
	}

	public void addQuestion(final Question qst) {
		QstRepository.save(qst);
	}

	public void deleteQuestion(final long idQuestion) {
		QstRepository.delete(idQuestion);
	}

	public Question getQuestion(final long idQuestion) {
		return QstRepository.findOne(idQuestion);
	}
}

File: src/main/resources/client/scripts/question/questionService.js
angular.module('app')
    .factory('questionService', ['$http', function ($http) {

        var urlBase = 'http://localhost:8090/question';
        var dataFactory = {};

        dataFactory.getAll = function () {
            return $http.get(urlBase);
        };

        dataFactory.getQuestion = function (id) {
            return $http.get(urlBase +'/'+id);
        };

        dataFactory.addQuestion = function (entity) {
            return $http.post(urlBase,entity);
        };

        dataFactory.updateQuestion = function (entity) {
            return $http.put(urlBase,entity)
        };

        dataFactory.deleteQuestion = function (id) {
            return $http.delete(urlBase + '/delete/' + id);
        };
        return dataFactory;
    }]);


